tests/Functional/Types/BigIntTypeTest.php

```diff
@@ -0,0 +1,121 @@
+
+        $this->connection->executeStatement(<<<SQL
+            INSERT INTO bigint_type_test (id, my_integer)
+            VALUES (42, $sqlLiteral)
+            SQL);
+
+        self::assertSame(
+            $expectedValue,
+            $this->connection->convertToPHPValue(
+                $this->connection->fetchOne('SELECT my_integer from bigint_type_test WHERE id = 42'),
+                Types::BIGINT,
+            ),
+        );
+    }
+
+    /** @return Generator<string, array{string, int|string|null}> */
+    public static function provideBigIntLiterals(): Generator
+    {
+        yield 'zero' => ['0', 0];
+        yield 'null' => ['null', null];
+        yield 'positive number' => ['42', 42];
+        yield 'negative number' => ['-42', -42];
+
+        if (PHP_INT_SIZE < 8) {
+            // The following tests only work on 64bit systems.
+            return;
+        }
+
+        // PHP_INT_MAX - 1
+        yield 'large positive number' => ['9223372036854775806', 9223372036854775806];
+
+        // PHP_INT_MIN + 1
+        yield 'large negative number' => ['-9223372036854775807', -9223372036854775807];
+    }
+
+    #[DataProvider('provideBigIntEdgeLiterals')]
+    public function testSelectBigIntEdge(int $value): void
+    {
+        $table = new Table('bigint_type_test');
+        $table->addColumn('id', Types::SMALLINT, ['notnull' => true]);
+        $table->addColumn('my_integer', Types::BIGINT, ['notnull' => false]);
+        $table->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($table);
+
+        $this->connection->executeStatement(<<<SQL
+            INSERT INTO bigint_type_test (id, my_integer)
+            VALUES (42, $value)
+            SQL);
+
+        self::assertThat(
+            $this->connection->convertToPHPValue(
+                $this->connection->fetchOne('SELECT my_integer from bigint_type_test WHERE id = 42'),
+                Types::BIGINT,
+            ),
+            LogicalXor::fromConstraints(new IsIdentical($value), new IsIdentical((string) $value)),
+        );
+    }
+
+    /** @return Generator<string, array{int}> */
+    public static function provideBigIntEdgeLiterals(): Generator
+    {
+        yield 'max int' => [PHP_INT_MAX];
+        yield 'min int' => [PHP_INT_MIN];
+    }
+
+    public function testUnsignedBigIntOnMySQL(): void
+    {
+        if (! TestUtil::isDriverOneOf('mysqli', 'pdo_mysql')) {
+            self::markTestSkipped('This test only works on MySQL/MariaDB.');
+        }
+
+        $table = new Table('bigint_type_test');
+        $table->addColumn('id', Types::SMALLINT, ['notnull' => true]);
+        $table->addColumn('my_integer', Types::BIGINT, ['notnull' => false, 'unsigned' => true]);
+        $table->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($table);
+
+        // Insert (2 ** 64) - 1
+        $this->connection->executeStatement(<<<'SQL'
+            INSERT INTO bigint_type_test (id, my_integer)
+            VALUES (42, 18446744073709551614)
```

Comment:
This cannot be 2<sup>64</sup> - 1 since it's an even number. Similar to the above, let's code it as an expression instead of a magic literal.